stacks/tcpconn.go

```diff
@@ -414,14 +420,15 @@ func (sock *TCPConn) send(response []byte) (n int, err error) {
 	var payload []byte
 	if available > 0 {
 		payload = response[sizeTCPNoOptions : sizeTCPNoOptions+seg.DATALEN]
+		//we are reading out of the TX ring buffer, data to encapsulate and send 
 		n, err = sock.tx.Read(payload)
 		if err != nil && err != io.EOF || n != int(seg.DATALEN) {
-			panic("bug in handleUser") // This is a bug in ring buffer or a race condition.
+			panic("bug in handleUser") // This is a bug in ring buffer or a race condition. - is it ? - odd message then
```

Comment:
We can do with a better message, maybe something like: `unexpected condition in seqs.TCPConn.send`